crates/loam-cli/src/commands/dev/mod.rs

```diff
@@ -0,0 +1,103 @@
+    Watcher(#[from] notify::Error),
+    #[error(transparent)]
+    Build(#[from] build::Error),
+    #[error("IO error: {0}")]
+    Io(#[from] std::io::Error),
+}
+
+impl Cmd {
+    pub async fn run(&self) -> Result<(), Error> {
+        // Set LOAM_ENV to development
+        std::env::set_var("LOAM_ENV", "development");
+        let (tx, mut rx) = tokio::sync::mpsc::channel(100);
+        let tx_clone = tx.clone();
+        let mut watcher =
+            notify::recommended_watcher(move |res: Result<notify::Event, notify::Error>| {
+                if let Ok(event) = res {
+                    if let notify::EventKind::Modify(_) = event.kind {
+                        if let Some(path) = event.paths.first() {
+                            println!("File modified: {:?}", path);
+                            // Send a signal through the channel to trigger a rebuild
+                            if let Err(e) = tx_clone.blocking_send(()) {
+                                eprintln!("Error sending through channel: {}", e);
+                            }
+                        }
+                    }
+                }
+            })
+            .unwrap();
+
+        let workspace_root = self
+            .build_cmd
+            .manifest_path
+            .parent()
+            .map(|path| path.to_path_buf())
+            .unwrap_or_else(|| PathBuf::from("."));
+        let env_toml = workspace_root.join("environments.toml");
+        if env_toml.exists() {
+            watcher.watch(
+                workspace_root.join("environments.toml").as_path(),
+                RecursiveMode::NonRecursive,
+            )?;
+        }
+        let packages = self.build_cmd.list_packages()?;
+        for package in packages {
+            let package_path = Path::new(package.manifest_path.parent().unwrap().as_str());
+            watcher.watch(package_path, RecursiveMode::Recursive)?;
+        }
+        if let Err(e) = self.build().await {
+            eprintln!("Build error: {}", e);
+        }
+        println!("Watching for changes. Press Ctrl+C to stop.");
+
+        loop {
+            tokio::select! {
+                _ = rx.recv() => {
+                    println!("Changes detected. Rebuilding...");
+                    if let Err(e) = self.build().await {
+                        eprintln!("Build error: {}", e);
+                    }
+                }
+                _ = tokio::signal::ctrl_c() => {
+                    println!("Stopping dev mode.");
+                    break;
+                }
+            }
+
+            // Debounce to avoid multiple rapid rebuilds
+            time::sleep(std::time::Duration::from_secs(1)).await;
+        }
+
+        Ok(())
+    }
+
+    async fn build(&self) -> Result<(), Error> {
+        let mut build_cmd = self.build_cmd.clone();
+        build_cmd
+            .build_clients
+            .env
+            .get_or_insert(LoamEnv::Development);
+        build_cmd.profile = "dev".to_string();
```

Comment:
```suggestion
        build_cmd.profile = "debug".to_string();
```

Reply:
Seems to not like using the name debug: 
```
error: profile name `debug` is reserved
To configure the default development profile, use the name `dev` as in [profile.dev]
```

Reply:
Ah yeah. `cargo build` is debug by default, you need to add either `--release` or a `--profile <defined in Cargo.toml>`. So this is more reason to make this an option and then it will be debug by default.

Reply:
I see. I changed it to be an option and default to debug. If its debug it doesnt add the --profile command.